aikau/src/main/resources/alfresco/services/actions/NodeLocationService.js

```diff
@@ -0,0 +1,163 @@
+ * {
+ *    name: "alfresco/services/actions/NodeLocationService",
+ *    config: {
+ *       nonSiteUrl: "custom-repository",
+ *       siteUrl: null
+ *    }
+ * }
+ *
+ * @example <caption>Configuration for standalone Aikau client:</caption>
+ * {
+ *    name: "alfresco/services/actions/NodeLocationService",
+ *    config: {
+ *       useAikauPages: true,
+ *       siteUrl: "doclib",
+ *       nonSiteUrl: "repo"
+ *    }
+ * }
+ *
+ * @module alfresco/services/actions/NodeLocationService
+ * @extends module:alfresco/core/Core
+ * @mixes module:alfresco/core/UrlUtils
+ * @author Dave Draper
+ */
+define(["dojo/_base/declare",
+        "alfresco/core/Core",
+        "alfresco/core/UrlUtils",
+        "dojo/_base/lang"],
+        function(declare, AlfCore, UrlUtils, lang) {
+
+   return declare([AlfCore, UrlUtils], {
+
+      /**
+       * The should be configured to be the URL fragment to be used when linking to nodes that are not located
+       * within a site. It defaults to "repository" as this is the standard non-site URL expected to be used within
+       * Alfresco Share.
+       * 
+       * @instance
+       * @type {string}
+       * @default "repository"
+       */
+      nonSiteUrl: "repository",
+
+      /**
+       * It's possible to explicitly set a specific URI template to use for generating links. If this is not
+       * set then a default of "sitepage" is used which is known to be configured in in both Alfresco Share
+       * and in applications generated from the Aikau Maven archetype.
+       *
+       * @instance
+       * @type {string}
+       * @default null
+       */
+      sitePageTemplate: null,
+
+      /**
+       * The should be configured to be the URL fragment to be used when linking to nodes that located
+       * within a site. It defaults to "documentlibrary" as this is the standard site URL expected to be used within
+       * Alfresco Share. If this is configured as being null then the 
+       * nonSiteUrl]{@link module:alfresco/services/actions/NodeLocationService#nonSiteUrl} will be used
+       * to as the location of all items.
+       * 
+       * @instance
+       * @type {string}
+       * @default "documentlibrary"
+       */
+      siteUrl: "documentlibrary",
+
+      /**
+       * This indicates whether or not the linked pages are "Aikau-style" page. An Aikau style page is where
+       * the a combination of "page-id" and "ws" URL tokens are used to construct a page. This defaults to false
+       * for use in Alfresco Share where pages for located items are typically standard "Surf-style" pages.
+       *
+       * @instance
+       * @type {boolean}
+       * @default false
+       */
+      useAikauPages: false,
+
+      /**
+       * @instance
+       * @param {array} args Constructor arguments
+       */
+      constructor: function alfresco_services_actions_NodeLocationService__constructor(args) {
+         lang.mixin(this, args);
```

Comment:
Not sure, I've used lang.mixin in all the other service constructors as well. We'd probably want to change all of them.